homeassistant/components/simplepush/notify.py

```diff
@@ -44,45 +54,120 @@ async def async_get_service(
         )
         return None
 
-    return SimplePushNotificationService(discovery_info)
+    return SimplePushNotificationService(hass, discovery_info)
 
 
 class SimplePushNotificationService(BaseNotificationService):
     """Implementation of the notification service for Simplepush."""
 
-    def __init__(self, config: dict[str, Any]) -> None:
+    def __init__(self, hass: HomeAssistant, config: dict[str, Any]) -> None:
         """Initialize the Simplepush notification service."""
+        self.hass = hass
+        self.session = async_get_clientsession(hass)
         self._device_key: str = config[CONF_DEVICE_KEY]
         self._event: str | None = config.get(CONF_EVENT)
         self._password: str | None = config.get(CONF_PASSWORD)
         self._salt: str | None = config.get(CONF_SALT)
 
-    def send_message(self, message: str, **kwargs: Any) -> None:
+    async def async_send_message(self, message: str, **kwargs: Any) -> None:
         """Send a message to a Simplepush user."""
         title = kwargs.get(ATTR_TITLE, ATTR_TITLE_DEFAULT)
 
+        actions = None
+        action_ids = {}
+        attachments = None
         # event can now be passed in the service data
         event = None
+        feedback_action_timeout = None
+
         if data := kwargs.get(ATTR_DATA):
             event = data.get(ATTR_EVENT)
 
+            actions_data = data.get(ATTR_ACTIONS)
+            if isinstance(actions_data, list) and actions_data:
+                actions = []
+                for action in actions_data:
+                    if "action" in action and "url" in action:
+                        actions.append({"name": action["action"], "url": action["url"]})
+                    elif "action" in action:
+                        actions.append(action["action"])
+                        if "id" in action:
+                            action_ids.update({action["action"]: action["id"]})
+
+                try:
+                    feedback_action_timeout = int(
+                        data.get(ATTR_FEEDBACK_ACTION_TIMEOUT)
+                    )
+                except (ValueError, TypeError):
+                    feedback_action_timeout = 60
+
+            attachments_data = data.get(ATTR_ATTACHMENTS)
```

Comment:
Please limit the PR to one thing. Don't both add callbacks and attachments.

Reply:
Okay, I'll remove attachments from this PR and create a separate PR for them.